File: python/src/patterns/builder/product.py
class Product:
    """The complex object under construction."""

    def __init__(self) -> None:
        self._parts: list[str] = []

    def add(self, part: str) -> None:
        self._parts.append(part)

    def describe(self) -> str:
        return f"Product({', '.join(self._parts)})"

File: java/src/main/java/com/penapereira/patterns/builder/Product.java
package com.penapereira.patterns.builder;

import java.util.ArrayList;
import java.util.List;

/** The complex object under construction. */
public final class Product {

    private final List<String> parts = new ArrayList<>();

    public void add(String part) {
        parts.add(part);
    }

    public String describe() {
        return "Product(" + String.join(", ", parts) + ")";
    }
}
